src/codeu/chat/server/Server.java

```diff
@@ -60,13 +60,63 @@
   //Creates an instance of ServerInfo that helps keep the Time and version of when the server started
   private static ServerInfo serverInfo;
 
-  public Server(final Uuid id, final Secret secret, final Relay relay) {
+  //Log Files Info
+  private static String serverLogLocation = "C:\\git\\CodeU-Summer-2017\\serverdata\\serverLog.txt";
+  public PrintWriter outputStream;
+  LogReader logReader;
+
+  //Store the lines of Log
+  ArrayList<String> fileLines;
+
+  public Server(final Uuid id, final Secret secret, final Relay relay) throws IOException{
 
     this.id = id;
     this.secret = secret;
     this.controller = new Controller(id, model);
     this.relay = relay;
 
+
+    //Connects the Log to the Server
+    try {
+        outputStream = new PrintWriter(new FileWriter(serverLogLocation, true));
+    }catch (FileNotFoundException e){
+      e.printStackTrace();
+    }
+
+    //Set-ups OutputStream to append to current Log Information
+    outputStream.append("");
+    outputStream.flush();
+
+    //Reads in the Log and stores the lines in an ArrayList
+    logReader = new LogReader();
+    fileLines = logReader.readFile();
+
+    //Loads in the fileLines from the Log
+    for(int i = 0; i < fileLines.size(); i++){
+      LogLoader logLoader = new LogLoader(fileLines.get(i));
+
+      //Load in User
+      if(logLoader.findCommmand().equals("U")){
+        String[] userInfo = logLoader.loadUser();
+        this.controller.newUser(Uuid.parse(userInfo[0]), userInfo[1], Time.now());
+      }
+      //Load in Conversation
+      else if(logLoader.findCommmand().equals("C")){
+        String[] userInfo = logLoader.loadConversation();
+        this.controller.newConversation(Uuid.parse(userInfo[0]), userInfo[1], Uuid.parse(userInfo[2]), Time.fromMs(Long.parseLong(userInfo[3])));
```

Comment:
Well, a person could just look back to the `loadConversation()` where it's easily formatted. It explains and shows all of the values in userInfo. I thought it would be a bit messy to include it here as well.